components/script/dom/webgl2renderingcontext.rs

```diff
@@ -903,6 +974,168 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
                             level: i32) {
         self.base.FramebufferTexture2D(target, attachment, textarget, texture, level)
     }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.16
+    fn BindBufferBase(&self, target: u32, index: u32, buffer: Option<&WebGLBuffer>) {
+        let slot = match target {
+            constants::UNIFORM_BUFFER => &self.bound_uniform_buffer,
+            constants::TRANSFORM_FEEDBACK_BUFFER => &self.bound_transform_feedback_buffer,
+            // Invalid parameters
+            _ => return self.webgl_error(InvalidEnum)
+        };
+
+        if let Some(error) = buffer.and_then(|buffer| buffer.update_target(target).err()) {
+            return self.webgl_error(error);
+        }
+        WebGLBuffer::update_slot(slot, buffer);
+
+        self.send_command(WebGLCommand::BindBufferBase(target, index, buffer.map(|b| b.id())));
+    }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.16
+    fn BindBufferRange(&self, target: u32, index: u32, buffer: Option<&WebGLBuffer>, offset: i64, size: i64) {
+        let slot = match target {
+            constants::UNIFORM_BUFFER => &self.bound_uniform_buffer,
+            constants::TRANSFORM_FEEDBACK_BUFFER => &self.bound_transform_feedback_buffer,
+            // Invalid parameters
+            _ => return self.webgl_error(InvalidEnum)
+        };
+
+        if let Some(error) = buffer.and_then(|buffer| buffer.update_target(target).err()) {
+            return self.webgl_error(error);
+        }
+        WebGLBuffer::update_slot(slot, buffer);
+
+        self.send_command(WebGLCommand::BindBufferRange(target, index, buffer.map(|b| b.id()), offset, size));
+    }
+
+    #[allow(unsafe_code)]
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.16
+    unsafe fn GetIndexedParameter(&self, cx: *mut JSContext, target: u32, index: u32) -> JSVal {
+        let value = match target {
+            constants::TRANSFORM_FEEDBACK_BUFFER_BINDING =>
+                return object_binding_to_js_or_null!(cx, &self.bound_transform_feedback_buffer),
+            constants::UNIFORM_BUFFER_BINDING =>
```

Comment:
These should be returning the indexed bound buffer, not the non-indexed binding's buffer.  We would need to track a vec of bindings in BindBufferBase/Range according to the context limits, and also validate that the index is not greater than that array size, and then return them here.